NDTensors/src/dense/dense.jl

```diff
@@ -20,38 +20,43 @@ struct Dense{ElT,VecT<:AbstractArray} <: TensorStorage{ElT}
   end
 end
 
-function Dense(data::VecT) where {VecT<:AbstractArray{ElT}} where {ElT}
-  return Dense{ElT,VecT}(data)
+## We need a general typeinfo constructor for Dense
+default_Densetype(datatype::Type{<:AbstractArray{ElT}}) where {ElT} = default_storagetype(datatype)
```

Comment:
Looking through the code below, I don't think this function is really necessary, and the same functionality can be implemented with a combination of the `default_datatype` function and the `Dense` constructor.